src/components/CommitteeTools/components/Decrypt/Decrypt.tsx

```diff
@@ -7,136 +7,170 @@ import { decryptKey, readPrivateKey } from "openpgp";
-  return passphrase ? await decryptKey({
-    privateKey: await readPrivateKey({ armoredKey: privateKey }),
-    passphrase
-  }) : await readPrivateKey({ armoredKey: privateKey })
+import CopyIcon from "assets/icons/copy.icon.svg";
+
+export async function readPrivateKeyFromStoredKey(
+  privateKey: string,
+  passphrase: string | undefined
+) {
+  return passphrase
+    ? await decryptKey({
+        privateKey: await readPrivateKey({ armoredKey: privateKey }),
+        passphrase
+      })
+    : await readPrivateKey({ armoredKey: privateKey });
 }
 
-
 export default function Decrypt() {
-  const vaultContext = useContext(VaultContext)
-  const [showSelectKeyModal, setShowSelectKeyModal] = useState(false)
-  const [showSelectedKeyDetails, setShowSelectedKeyDetails] = useState(false)
+  const vaultContext = useContext(VaultContext);
+  const [showSelectKeyModal, setShowSelectKeyModal] = useState(false);
+  const [showSelectedKeyDetails, setShowSelectedKeyDetails] = useState(false);
   const [error, setError] = useState<string>();
-  const encryptedMessageRef = useRef<HTMLDivElement>(null);
-  const decryptedMessageRef = useRef<HTMLDivElement>(null);
+  const encryptedMessageRef = useRef<HTMLTextAreaElement>(null);
+  const decryptedMessageRef = useRef<HTMLTextAreaElement>(null);
   const { t } = useTranslation();
 
-
-  const location = useLocation()
+  const location = useLocation();
 
   useEffect(() => {
     const urlSearchParams = new URLSearchParams(location.search);
     const params = Object.fromEntries(urlSearchParams.entries());
     if (params.ipfsjson) {
       (async () => {
-        const response = await fetch(params.ipfsjson)
-        const json = await response.json()
-        encryptedMessageRef.current!.textContent = json.message
-
-      })()
-
+        const response = await fetch(params.ipfsjson);
+        const json = await response.json();
+        encryptedMessageRef.current!.value = json.message;
+      })();
     }
-
-  }, [location.search])
+  }, [location.search]);
 
   useEffect(() => {
-    if (vaultContext.vault?.storedKeys.length === 0 || !vaultContext.selectedKey === undefined) {
-      setShowSelectKeyModal(true)
+    if (
+      vaultContext.vault?.storedKeys.length === 0 ||
+      !vaultContext.selectedKey === undefined
+    ) {
+      setShowSelectKeyModal(true);
     }
-  }, [vaultContext.vault, vaultContext.selectedKey])
+  }, [vaultContext.vault, vaultContext.selectedKey]);
 
   const _decrypt = useCallback(async () => {
     try {
       if (!vaultContext.selectedKey) {
-        setShowSelectKeyModal(true)
-        return
+        setShowSelectKeyModal(true);
+        return;
       }
 
-      const armoredMessage = encryptedMessageRef.current!.textContent
+      const armoredMessage = encryptedMessageRef.current!.value;
 
       if (!armoredMessage || armoredMessage === "") {
-        throw new Error("No message to decrypt")
+        throw new Error("No message to decrypt");
```

Comment:
Please use the text from the translation file of `i18next`. If necessary let's add an errors section there.

Reply:
Of course, will do